xbmc/guilib/XBTFReader.cpp

```diff
@@ -31,16 +31,16 @@
 #include "PlatformDefs.h"
 
 #define READ_STR(str, size, file) \
-  if (!fread(str, size, 1, file)) \
+  if (1 != fread(str, size, 1, file)) \
```

Comment:
Yes, you are right - it will do what we want in this situation, where we always want to read 1 element.